packages/expo-cli/src/commands/google-play/AppSigningOptIn.js

```diff
@@ -0,0 +1,254 @@
+  };
+
+  // Keystore used to sign production app
+  signKeystore: string = '';
+  // Keystore used to sign app before uploading to Google Play store
+  uploadKeystore: string = '';
+  // private signing key and public cert extracted from signKeystore and encrypted using Google Play encryption key.
+  privateSigningKey: string = '';
+  // public cert extracted from upload keystore
+  publicUploadCert: string = '';
+
+  uploadKeystoreCredentials: Object = {};
+  signKeystoreCredentials: Object = {};
+
+  constructor(projectDir: string, options: Options) {
+    this.projectDir = projectDir;
+    this.options = options;
+  }
+
+  async run() {
+    const { username } = (await User.ensureLoggedInAsync()) || {};
+    const { exp = {} } = await ProjectUtils.readConfigJsonAsync(this.projectDir);
+
+    await this.init(exp.slug);
+
+    log(`Saving current keystore to ${this.signKeystore}...`);
+    this.signKeystoreCredentials = await Credentials.Android.backupExistingCredentials({
+      outputPath: this.signKeystore,
+      username,
+      experienceName: `@${username}/${exp.slug}`,
+    });
+
+    await this.exportPrivateKey();
+
+    if (this.options.uploadKey) {
+      await this.runWithUploadCert(username, exp);
+    } else {
+      await this.runWithoutUploadCert();
+    }
+  }
+
+  async init(slug: string) {
+    log.warn(
+      'Make sure that you are not using google play signing already, this process will remove current keystore from expo servers.'
+    );
+    log(
+      `You can check it here ${chalk.underline(
+        'https://play.google.com/apps/publish'
+      )}, select app and go to \"Release managment\" -> \"App signing\" tab, if you are already using google play signing there will be a message \"App Signing by Google Play is enabled for this app\" at the top of the page.`
+    );
+    const confirmQuestion = [
+      { type: 'confirm', name: 'confirm', message: 'Is Google Play signing enabled for this app?' },
+    ];
+    const { confirm: confirmEnabled } = await prompt(confirmQuestion);
+    if (confirmEnabled) {
+      log('Google Play signing already enabled, there is nothing to do here.');
+      process.exit(0);
+    }
+
+    this.signKeystore = path.join(this.projectDir, `${slug}_sign.jks.bak`);
+    this.uploadKeystore = path.join(this.projectDir, `${slug}_upload.jks.tmp`);
+    this.privateSigningKey = path.join(this.projectDir, `${slug}_private_sign_key`);
+    this.publicUploadCert = path.join(this.projectDir, `${slug}_upload_cert.pem`);
+    await this.cleanup(true);
+  }
+
+  async exportPrivateKey() {
+    log(
+      `Go to "App signing" tab in Google Play console, select "${chalk.bold(
+        'Export and upload a key (not using a Java keystore)'
+      )}" and copy encryption key that is listed in step 1.`
+    );
+    const encryptKeyQuestion = [
+      { type: 'input', name: 'encryptionKey', message: 'Google Play encryption key' },
+    ];
+    const { encryptionKey } = await prompt(encryptKeyQuestion);
```

Comment:
`prompt` is just a wrapper for https://github.com/enquirer/enquirer You can avoid the following `if` by passing `validate` function in the question object.